File: apps/web/lib/api/watchlist.ts
import { authFetch, readError } from "./auth-fetch";

export type WatchlistItem = {
  instrument_id: string;
  added_at: string;
  symbol: string;
  exchange: string;
  name: string;
  asset_class: string;
  currency: string;
  price: number;
  change_pct: number;
};

export async function listWatchlist(): Promise<WatchlistItem[]> {
  const res = await authFetch("/v1/watchlist");
  if (!res.ok) throw await readError(res);
  return res.json() as Promise<WatchlistItem[]>;
}

export async function addWatchlist(instrument_id: string): Promise<void> {
  const res = await authFetch("/v1/watchlist", {
    method: "POST",
    body: JSON.stringify({ instrument_id }),
  });
  if (!res.ok && res.status !== 201) throw await readError(res);
}

export async function removeWatchlist(instrument_id: string): Promise<void> {
  const res = await authFetch(`/v1/watchlist/${instrument_id}`, {
    method: "DELETE",
  });
  if (!res.ok && res.status !== 204) throw await readError(res);
}

File: apps/api/internal/models/watchlist.go
package models

import "time"

type WatchlistItem struct {
	InstrumentID string    `json:"instrument_id"`
	AddedAt      time.Time `json:"added_at"`
	Symbol       string    `json:"symbol"`
	Exchange     string    `json:"exchange"`
	Name         string    `json:"name"`
	AssetClass   string    `json:"asset_class"`
	Currency     string    `json:"currency"`
	Price        float64   `json:"price"`      // quotes.price, 없으면 0
	ChangePct    float64   `json:"change_pct"`
}
